pkg/app/app.go

```diff
@@ -66,6 +70,86 @@ func GetFromSlug(slug string) (*App, error) {
 	return Get(id)
 }
 
+func GetLicenseDataFromDatabase(id string) (string, error) {
+	logger.Debug("getting app license from database",
+		zap.String("id", id))
+
+	db := persistence.MustGetPGSession()
+	query := `select license from app where id = $1`
+	row := db.QueryRow(query, id)
+
+	license := ""
+
+	if err := row.Scan(&license); err != nil {
+		return "", errors.Wrap(err, "failed to scan license")
+	}
+
+	return license, nil
+}
+
+func Create(name string, upstreamURI string, licenseData string, isAirgapEnabled bool) (*App, error) {
+	logger.Debug("creating app",
+		zap.String("name", name),
+		zap.String("upstreamURI", upstreamURI))
+
+	db := persistence.MustGetPGSession()
+	tx, err := db.Begin()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to begin transaction")
+	}
+	defer tx.Rollback()
+
+	titleForSlug := strings.Replace(name, ".", "-", 0)
+	slugProposal := slug.Make(titleForSlug)
+
+	foundUniqueSlug := false
+	i := 0
+	for !foundUniqueSlug {
+		if i > 0 {
+			slugProposal = fmt.Sprintf("%s-%d", titleForSlug, i)
+		}
+
+		query := `select count(1) as count from app where slug = $1`
+		row := tx.QueryRow(query, slugProposal)
+		exists := 0
+		if err := row.Scan(&exists); err != nil {
+			return nil, errors.Wrap(err, "failed to scan existing slug")
+		}
+
+		if exists == 0 {
+			foundUniqueSlug = true
+		} else {
+			i++
+		}
+	}
+
+	installState := ""
+	if strings.HasPrefix(upstreamURI, "replicated://") == false {
+		installState = "installed"
+	} else {
+		if isAirgapEnabled {
+			installState = "airgap_upload_pending"
+		} else {
+			installState = "online_upload_pending"
```

Comment:
wouldn't `replicated://` fall into this  category? this might need a comment.

Reply:
Yeah, the if else statement covers this?